Makefile.gccm68k

```diff
@@ -0,0 +1,79 @@
+INTERP_SOURCEFILES =  Interpreter.o Interpreter_Error.o Interpreter_PrimArith.o
+SNAP_SOURCEFILES = ObjectMemory_Snapshot.c
+
+# #########################################################
+# Header files for VM - ObjectMemory, Interpreter and Snapshot
+OM_HEADERFILES = ObjectMemory.h ObjectMemory_Constants.h ObjectMemory_Storage.h ObjectMemory_FreeList.h ObjectMemory_Allocation.h ObjectMemory_RefCounting.h ObjectMemory_GC.h RealWordMemory.h 
+INTERP_HEADERFILES = Interpreter.h Interpreter_Constants.h Interpreter_PrimArith.h Interpreter_Error.h 
+SNAP_HEADERFILES = ObjectMemory_Snapshot.h
+
+# #########################################################
+# Shared Header files 
+SHARED_HEADERFILES = Types.h
+
+# #########################################################
+# Source and Headers for the unit tests
+TEST_SOURCES = test_main.o FreeListTests.o ObjectMemoryTests.o RealWordMemoryTests.o AllocationTests.o RefCountingTests.o PrimitiveMethodTests.o InterpreterMemoryTests.o Interpreter_IntegerPrimTests.o Interpreter_IntegerArithTests.o Interpreter_FloatArithTests.o Interpreter_IntegerCmpTests.o CompiledMethodTests.o ContextTests.o ClassTests.o
+TEST_HEADERS = tests.h FreeListTests.h ObjectMemoryTests.h RealWordMemoryTests.h AllocationTests.h RefCountingTests.h PrimitiveMethodTests.h InterpreterMemoryTests.h Interpreter_IntegerPrimTests.h Interpreter_IntegerArithTests.h Interpreter_FloatArithTests.h Interpreter_IntegerCmpTests.h CompiledMethodTests.h ContextTests.h ClassTests.h
+
+# #########################################################
+# Source and Headers the small error host
+ERROR_SOURCES = Interpreter_Error.o Error_Host.o
+ERROR_HEADERS = Interpreter_Error.h
+
+# #########################################################
+# O files come from C files
+%.o: %.c
+	$(CC) -c -o $@ $< $(CFLAGS)
+
+# #########################################################
+# Build the various binaries
+$(TESTAPP): $(OM_SOURCEFILES)  $(INTERP_SOURCEFILES)  $(OM_HEADERFILES) $(INTERP_HEADERFILES) $(SHARED_HEADERFILES) $(TEST_SOURCES) $(TEST_HEADERS)
+	$(CC) -o $@ $(OM_SOURCEFILES)  $(INTERP_SOURCEFILES) $(TEST_SOURCES) $(CFLAGS) $(LDFLAGS)
+
+$(SNAPAPP): $(OM_SOURCEFILES)  $(INTERP_SOURCEFILES) $(SNAP_SOURCEFILES) $(OM_HEADERFILES) $(INTERP_HEADERFILES) $(SNAP_HEADERFILES) $(SHARED_HEADERFILES)
+	$(CC) -o $@ $(OM_SOURCEFILES)  $(INTERP_SOURCEFILES) $(SNAP_SOURCEFILES) $(CFLAGS) $(LDFLAGS)
+
+$(ERRORHOST): $(ERROR_SOURCES) $(ERROR_HEADERS)
+	$(CC) -o $@ $(ERROR_SOURCES) $(CFLAGS) $(LDFLAGS)
+
+all: $(TESTAPP) $(SNAPAPP) $(ERRORHOST)
+
+# #########################################################
+# Cleanup
+clean:
+	rm -f *.o
+	rm -f $(TESTAPP)
+	rm -f $(SNAPAPP)
+	rm -f $(ERRORHOST)
+
+# #########################################################
+# Run the unit tests inside vamos
+runtests: clean all 
+	vamos -V System:$(AMIGASYS) -s 16 -m 8192  -- $(TESTAPP) -quiet
```

Comment:
Vamos needs an amiga system disk, anything you have installed on UAE is fine.  I also tell it to set the stack to 16K and the memory to 8Mb.  I'm passing `-quiet` to the ast_tests program to keep the noise down.  See elsewhere.